swagger_zipkin/otel_decorator.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import importlib
+from contextlib import contextmanager
+from typing import Any
+from typing import TYPE_CHECKING
+from typing import TypeVar
+
+from opentelemetry import trace
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+from opentelemetry.trace.span import format_span_id
+from opentelemetry.trace.span import format_trace_id
+from opentelemetry.trace.span import TraceFlags
+from typing_extensions import ParamSpec
+
+from swagger_zipkin.decorate_client import Client
+from swagger_zipkin.decorate_client import decorate_client
+from swagger_zipkin.decorate_client import Resource
+
+if TYPE_CHECKING:
+    import pyramid.request.Request   # type: ignore # noqa: F401
+
+T = TypeVar('T', covariant=True)
+P = ParamSpec('P')
+
+tracer = trace.get_tracer("otel_decorator")
+
+
+class OtelResourceDecorator:
+    """A wrapper to the swagger resource.
+
+    :param resource: A resource object. eg. `client.pet`, `client.store`.
+    :type resource: :class:`swaggerpy.client.Resource` or :class:`bravado_core.resource.Resource`
+    """
+
+    def __init__(self, resource: Client, client_identifier: str, smartstack_namespace: str) -> None:
+        self.resource = resource
+        self.client_identifier = client_identifier
+        self.smartstack_namespace = smartstack_namespace
+
+    def __getattr__(self, name: str) -> Resource:
+        return decorate_client(self.resource, self.with_headers, name)
```

Comment:
need help. static type checking is failing here: https://github.com/Yelp/swagger_zipkin/actions/runs/11841603555/job/32998236954?pr=19
The signatures are the same as the classes used in zipkin_decorator.py

Reply:
The errors are inside `decorate_client`. I suspect this was already failing.

Reply:
Fixed some signatures in all 3 files: decorate_client.py, zipkin_decorator.py, otel_decorator.py
Although i'm not sure why this error is showing up now as after adding an otel_decorator.py